spec/built-in-modules/math.md

```diff
@@ -2,107 +2,476 @@
+    * [`div()`](#div)
+    * [`percentage()`](#percentage)
+    * [`random()`](#random)
+
+## Variables
+
+### `$e`
+
+Equal to the value of the mathematical constant `e` with a precision of 10
+digits after the decimal point: `2.7182818285`.
+
+### `$pi`
+
+Equal to the value of the mathematical constant `pi` with a precision of 10
+digits after the decimal point: `3.1415926536`.
 
 ## Functions
 
-### `abs()`
-
-```
-abs($number)
-```
-
-This function is also available as a global function named `abs()`.
+### Bounding Functions
 
-### `ceil()`
+#### `ceil()`
 
 ```
 ceil($number)
 ```
 
 This function is also available as a global function named `ceil()`.
 
-### `compatible()`
+#### `clamp()`
 
 ```
-compatible($number1, $number2)
+clamp($min, $number, $max)
 ```
 
-This function is also available as a global function named `comparable()`.
+* If some arguments have units and some do not, throw an error.
+* If `$min`, `$number`, and `$max` have units, but the units are not
+  [compatible] with each other, throw an error.
+* If `$min >= $max`, return `$min`.
+* If `$number <= $min`, return `$min`.
+* If `$number >= $max`, return `$max`.
+* Return `$number`.
 
-### `floor()`
+[compatible]: ../types/number.md#compatible-units
+
+#### `floor()`
 
 ```
 floor($number)
 ```
 
 This function is also available as a global function named `floor()`.
 
-### `max()`
+#### `max()`
 
 ```
 max($numbers...)
 ```
 
 This function is also available as a global function named `max()`.
 
-### `min()`
+#### `min()`
 
 ```
 min($numbers...)
 ```
 
 This function is also available as a global function named `min()`.
 
-### `percentage()`
+#### `round()`
 
 ```
-percentage($number)
+round($number)
 ```
 
-This function is also available as a global function named `percentage()`.
+This function is also available as a global function named `round()`.
+
+### Distance Functions
 
-### `random()`
+#### `abs()`
 
 ```
-random($limit: null)
+abs($number)
 ```
 
-This function is also available as a global function named `random()`.
+This function is also available as a global function named `abs()`.
 
-### `round()`
+#### `hypot()`
 
 ```
-round($number)
+hypot($numbers...)
 ```
 
-This function is also available as a global function named `round()`.
+* If some numbers have units and some do not, throw an error.
+* If all numbers have units, but the units are not [compatible] with each other,
+  throw an error.
+* If all numbers are unitless, the return value is unitless.
+* Otherwise, the return value takes the unit of the leftmost number.
+* If any number equals `Infinity` or `-Infinity`, return `Infinity`.
+* Return the square root of the sum of the squares of each number.
+
+### Exponential Functions
+
+Exponential operations on numbers with units would create meaningless units
+(e.g. `(1px)^(1/3)` has a unit of `px^(1/3)`). To prevent this, the exponential
+functions accept only a unitless number as input, and return a unitless number.
```

Comment:
Since this is explanatory text, it should probably be in a block quote. Same with the explanatory text below.